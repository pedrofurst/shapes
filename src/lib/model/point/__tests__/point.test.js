import Point from '../index';

const DEFAULT_X = 10;
const DEFAULT_Y = 10;

describe('Point class', () => {

  it('Should instance the class and set the values', () => {
    const mockContext = {someContext: 'example'};
    const point = new Point(DEFAULT_X, DEFAULT_Y, mockContext);
    expect(point.x).toEqual(DEFAULT_X);
    expect(point.y).toEqual(DEFAULT_Y);
    expect(point.context).toEqual(mockContext);
  });

  it('Should validate if the user clicks above the point', () => {
    const point = new Point(DEFAULT_X, DEFAULT_Y);
    expect(point.isClickedAbove(5,5)).toBeTruthy();
    expect(point.isClickedAbove(50,50)).toBeFalsy();
  });

  it('Should draw the point', () => {
    const canvas = document.createElement("canvas");
    canvas.setAttribute("width", "200");
    canvas.setAttribute("height", "200");
    const ctx = canvas.getContext('2d');
    const point = new Point(DEFAULT_X, DEFAULT_Y, canvas.getContext('2d'));
    point.draw();
    expect(ctx.__getDrawCalls()).toMatchSnapshot();
  });
});
