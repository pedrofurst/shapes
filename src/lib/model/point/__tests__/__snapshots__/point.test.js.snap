// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Point class Should draw the point 1`] = `
Array [
  Object {
    "props": Object {
      "path": Array [
        Object {
          "props": Object {},
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        Object {
          "props": Object {
            "anticlockwise": false,
            "endAngle": 6.283185307179586,
            "radius": 0.4,
            "startAngle": 0,
            "x": 10,
            "y": 10,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "arc",
        },
      ],
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
  Object {
    "props": Object {
      "path": Array [
        Object {
          "props": Object {},
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        Object {
          "props": Object {
            "anticlockwise": false,
            "endAngle": 6.283185307179586,
            "radius": 5.5,
            "startAngle": 0,
            "x": 10,
            "y": 10,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "arc",
        },
      ],
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
  Object {
    "props": Object {
      "maxWidth": null,
      "text": "x: 10",
      "x": 10,
      "y": 30,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillText",
  },
  Object {
    "props": Object {
      "maxWidth": null,
      "text": "y: 10",
      "x": 10,
      "y": 45,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillText",
  },
]
`;
